tests/end_to_end/conftest.py

```
@@ -196,58 +214,46 @@ def fx_federation(request, pytestconfig):
+        f"\tNumber of collaborators: {request.config.num_collaborators}\n"
+        f"\tNumber of rounds: {request.config.num_rounds}\n"
+        f"\tModel name: {request.config.model_name}\n"
+        f"\tClient authentication: {request.config.require_client_auth}\n"
+        f"\tTLS: {request.config.use_tls}\n"
+        f"\tMemory Logs: {request.config.log_memory_usage}"
     )
 
     # Validate the model name and create the workspace name
-    if not model_name.upper() in constants.ModelName._member_names_:
-        raise ValueError(f"Invalid model name: {model_name}")
+    if not request.config.model_name.upper() in constants.ModelName._member_names_:
+        raise ValueError(f"Invalid model name: {request.config.model_name}")
 
-    workspace_name = f"workspace_{model_name}"
+    workspace_name = request.config.model_name
 
     # Create model owner object and the workspace for the model
-    model_owner = participants.ModelOwner(workspace_name, model_name, log_memory_usage)
+    model_owner = participants.ModelOwner(workspace_name, request.config.model_name, request.config.log_memory_usage)
     try:
-        workspace_path = model_owner.create_workspace(results_dir=results_dir)
+        workspace_path = model_owner.create_workspace(results_dir=request.config.results_dir)
     except Exception as e:
         log.error(f"Failed to create the workspace: {e}")
         raise e
 
     # Modify the plan
     try:
         model_owner.modify_plan(
-            new_rounds=num_rounds,
-            num_collaborators=num_collaborators,
-            require_client_auth=require_client_auth,
-            use_tls=use_tls,
+            new_rounds=request.config.num_rounds,
+            num_collaborators=request.config.num_collaborators,
+            require_client_auth=request.config.require_client_auth,
+            use_tls=request.config.use_tls,
         )
     except Exception as e:
         log.error(f"Failed to modify the plan: {e}")
         raise e
```

[FYR](https://google.github.io/styleguide/pyguide.html#24-exceptions), i.e. avoid catch-all exceptions as much as possible. Raising specific exceptions is encouraged within the function itself.

---

@noopurintel added this in next PR.